cmd/ingester/app/consumer/deadlock_detector.go

```diff
@@ -68,7 +68,8 @@ func newDeadlockDetector(metricsFactory metrics.Factory, logger *zap.Logger, int
 		time.Sleep(time.Second) // Allow time to flush metric
 
 		buf := make([]byte, 1<<20)
-		logger.Panic("No messages processed in the last check interval",
+		logger.Panic("No messages processed in the last check interval, possible deadlock, exiting. "+
```

Comment:
LGTM - only one suggestion might be to remove the stack trace, so it doesn't look like an unexpected exit?